### PR TITLE
Warnings about default value of  fileReader/fileWriter `locking` parameter changing

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
This PR implements a couple of warnings about the default values of `locking` changing in an upcoming release. The changes are planned to be:
* switch `locking`'s default value from true to false in `openReader` and `openWriter`
* remove `locking`'s default value from `file.reader` and `file.writer`

For both cases, the behavior of user code is not changed, only warnings are emitted. For `openReader`/`openWriter`, if `locking` isn't already specified, users will get a warning that the default is going to change in an upcoming release. They either need to specify locking explicitly or set `Open[Reader|Writer]LockingDefault=false` to opt in to the new default and silence the warning. For `reader`/`writer`, users will get a warning that leaving the value unspecified will eventually become an error — setting the value explicitly silences the warning.

The switch can be completed in a future release by:
* deprecating `OpenReaderLockingDefault` and `OpenWriterLockingDefault`, and setting the default values of `locking` to `false` in `openReader` and `openWriter`
* making the `reader`/`writer` case an error instead of a warning. This amounts to deleting an extra overload of each of these functions

**Related Issue:** https://github.com/chapel-lang/chapel/issues/24263

### Notes for reviewer:

The warnings are tested and shown in `test/deprecated/IO/lockingDefaultChange/`. 

Most of the changes are test updates that add `locking=false` to calls to the relevant procedures that didn't already specify a value for `locking`. Many of these modifications were made via a python-bindings script, but I've skimmed over them to make sure nothing was changed unexpectedly. In a few cases, I overwrote the changes with `locking=true` if the fileReader/Writer is used in a similar fashion to stdin/stdout (e.g., the c-ray tests, or `examples/benchmarks/lulesh`).

The only non-trivial code changes are in the IO module where the warnings themselves are implemented.

### Testing:

- [x] local paratest
- [x] gasnet paratest

### CODE REVIEW

**Comment by jeremiah-corrado** [comment]
Note: commits 2456583db6bded9d5b086f25c4ed83ef15832798..78a0fee296b974cba3a4df95af06c348234b77ca merged from https://github.com/chapel-lang/chapel/pull/24363 so I can begin working on test corrections before that PR gets merged.

**Review by lydia-duncan — COMMENTED** [review]
I went through some of this idly, but I don't think I can review the whole thing.  I had a couple of comments that I didn't want to leave hanging, though.

In addition to those, I think it would be useful to think about any `openReader`/`openWriter` tests from the perspective of "what do we want the test to do once the feature is in its new permanent state".  Put another way, I think it would be really easy to modify these test to avoid the warning and then just leave them modified, meaning that our test directory will then lack testing of the new default.  So I think for tests of `openReader`/`openWriter`, I'd be inclined to update the .good files to include the new warning messages instead of changing the test behavior, so that when the default is fully flipped, we'll just remove the warnings from the output and still have the same level of coverage of default/explicit behavior.  Most of the tests I was seeing in my initial pass were for changes to file.reader/file.writer instead, so hopefully that's not too much work for you, if you agree

**Comment by jeremiah-corrado** [comment]
> In addition to those, I think it would be useful to think about any openReader/openWriter tests from the perspective of "what do we want the test to do once the feature is in its new permanent state" ...  So I think for tests of openReader/openWriter, I'd be inclined to update the .good files to include the new warning messages instead of changing the test behavior, so that when the default is fully flipped, we'll just remove the warnings from the output and still have the same level of coverage of default/explicit behavior.

I share the same concern, but I was thinking about addressing it a different way. I had the script that I used to automatically modify the tests create a list of affected files. When it comes time to flip the default, I was planning to run the "inverse" of that script on those files, so that we'd end up with many uses of `openReader`/`openWriter` that do not specify a value for `locking`. Does that sound reasonable?

For the examples and primers that use `openReader`/`openWriter`, I can definitely see the benefit of including the warning in the `good` file (or perhaps compiling with `Open[Reader|Writer]LockingDefault=false`). I think I'll make those updates.

**Comment by lydia-duncan** [comment]
If you think you'll remember to run that script when the time comes to flip the default, I don't have a problem with that approach.  I just know it's easy to forget these things when time passes, even if you write a note for yourself.  But having 2 of us potentially remembering it helps with that, as does talking about it here.

File: test/release/examples/primers/specialMethods.chpl [hunk 1]
```diff
@@ -163,11 +163,10 @@ proc R.serialize(writer: fileWriter(?),
 }
 
 {
-  // Open the file in a new block so that deinitializers
+  // Open the fileReader in a new block so that deinitializers
```
Comment (lydia-duncan) [inline test/release/examples/primers/specialMethods.chpl:166]: I think you mean `fileWriter`

Comment (jeremiah-corrado) [reply test/release/examples/primers/specialMethods.chpl:166]: Yes, thank you!

**Review by lydia-duncan — APPROVED** [review]
Just some minor cleanup items, otherwise I think this is good to go!

File: test/library/packages/HDFS/hdfsNumbers.chpl.new [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
```
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply test/library/packages/HDFS/hdfsNumbers.chpl.new:1]: Yes, thanks for catching that!

**Pull request merged**